index.bs

```diff
@@ -4969,162 +4907,193 @@ The [=map/values=] of the [=context menu map=], which are
-
+To <dfn for=ContextMenus>[=dispatch=] a [=ContextMenus/show=] [=event=]</dfn>
+given a {{ContextMenus}} |contextMenus|, [=dispatch=] a [=new=] {{Event}}
+named [=ContextMenus/show=] to |contextMenus|.
 </div>
 
-{{ContextMenus}} has the attribute {{ContextMenus/onClicked}}, which is a {{ContextMenusEvent}} that is associated with [= click =] [= events =] on a [=context menu item=].
-
-<div algorithm='onclicked-event'>
-
-To [=dispatch=] an {{ContextMenus/onClicked}} [=event=] |event|, given {{HTMLControlledFrameElement}} |controlledFrame|, |element| as the {{HTMLElement}} inside the embedded document where the [=ContextMenus/context menu=] is opened at, |menuItem| as the [=context menu item=] that was clicked.
+<div algorithm>
+To <dfn for=ContextMenus>[=dispatch=] a [=ContextMenus/click=] [=event=]</dfn>
+given {{ContextMenus}} |contextMenus|, {{HTMLElement}} |element|, and |menuItem|
+as the [=context menu item=] that was clicked, run the following steps:
 
-1. Let |id| be the [=map/key=] of the [=ContextMenus/context menu map=]'s [=map/entry=] that is associated with |menuItem|.
+1. Let |id| be the [=map/key=] of the [=ContextMenus/context menu map=]'s
+    [=map/entry=] that is associated with |menuItem|.
 
-1. Let |properties| be the result of [=map/getting=] the [=map/value=] of the [=map/entry=] in [=ContextMenus/context menu map=] given the [=map/key=] |id|.
+1. Let |properties| be the result of [=map/getting=] the [=map/value=] of the
+    [=map/entry=] in [=ContextMenus/context menu map=] given the [=map/key=] |id|.
 
-1. Let |info| be a new {{OnClickData}}.
+1. Let |menuItem| be a [=new=] {{MenuItemDetails}}.
 
-    1. Update the following fields of |info|:
+    1. Update the following fields of |menuItem|:
 
-        : {{OnClickData/editable}}
+        : {{MenuItemDetails/editable}}
         :: Whether the |element| selected is editable, such as a text input.
 
-        : {{OnClickData/frameId}}
-        :: The [=navigable/frameId=] of the [=navigable container=] from which the [=/contextmenu=] [=event=] is dispatched.
+        : {{MenuItemDetails/frameId}}
+        :: The [=navigable/frameId=] of the [=navigable container=] from which
+            the [=/contextmenu=] [=event=] is dispatched.
 
-        : {{OnClickData/frameUrl}}
-        :: The [=Document/URL=] of the [= navigable container =] from which the [=/contextmenu=] [=event=] is dispatched.
+        : {{MenuItemDetails/frameURL}}
+        :: The [=Document/URL=] of the [=navigable container=] from which the
+            [=/contextmenu=] [=event=] is dispatched.
 
-        : {{OnClickData/menuItemId}}
+        : {{MenuItemDetails/id}}
         :: |id|.
 
-        : {{OnClickData/pageUrl}}
+        : {{MenuItemDetails/pageURL}}
         :: The [=Document/URL=] of the [=embedded navigable=].
 
-    1. Update the following optional fields of |info|:
+    1. Update the following optional fields of |menuItem|:
```

Comment:
These optional fields should be marked nullable in the definition of `MenuItemDetails`.